File: parentesi.c
#include <stdio.h>

int main( void ){
	printf( "stringa?\n" );
	int cont=0,pos=0;
	char ch;
	while( (ch=getchar())!='.' ){
		pos++;
		if( ch==')' )cont++;
		if( ch=='(' )cont--;
		if( cont>0 ){ printf("Parentesi chiusa di troppo in posizione %d\n",pos ); return 0;}
	}
	if( cont < 0 ) printf( "Ci sono parentesi aperte e non chiuse\n" );
	else printf( "Espressione ben parentesizzata\n" );
	return 0;
}

File: Parentesi.java
import java.io.*;

public class Parentesi{

	public static void main( String [] args ) throws IOException {
	
		BufferedReader in = new BufferedReader( new InputStreamReader( System.in ) );
		String str = in.readLine();
		
		int indic = 0;
		for( int i = 0; i < str.length(); i++ ){
			char corr = str.charAt(i);
			if( corr == ')' )indic++;
			if( corr == '(' )indic--;
			if( indic > 0 ){ 
				System.out.println("Parentesi chiusa di troppo in posizione " + (i+1) );
				return;
			}
		}	if( indic < 0 )
				System.out.println("Ci sono parentesi aperte e non chiuse" );
		
	
	}

}
